src/pages/api/auth/[...nextauth].ts

```diff
@@ -1,15 +1,30 @@
-import NextAuth from "next-auth";
+import NextAuth, { type NextAuthOptions, type DefaultSession } from "next-auth";
 import { PrismaAdapter } from "@next-auth/prisma-adapter";
 import GitHubProvider from "next-auth/providers/github";
 import prisma from "@/lib/prisma";
-
 import type { NextApiHandler } from "next";
 
+declare module "next-auth" {
+  interface Session extends DefaultSession {
+    user: {
+      id: string;
+    } & DefaultSession["user"];
```

Comment:
[ASK], why the typing can't be directly coming from the `DefaultSession` ya mas?